lib/state/ui/middleware.ts

```diff
@@ -30,9 +30,9 @@ export const middleware: Middleware<{}, S.State, Dispatch> = store => {
 
       // on updating the search field we should delay the update
       // so we don't waste our CPU time and lose responsiveness
-      case 'App.search':
+      case 'SEARCH':
         clearTimeout(searchTimeout);
-        if ('App.search' === action.type && !action.filter) {
+        if ('SEARCH' === action.type && !action.searchQuery) {
```

Comment:
due to the type narrowing we don't need to re-verify the action type. it suffices to test `if (!action.searchQuery)`

Reply:
Simplified in 7f0a867